### PR TITLE
Notify for app pause and resume events on Android

### LINKED ISSUE #43: Config GDScript Classes only valid after enter_vr_mode()
I just noticed that there is no check if the ovr mobile context is NULL at https://github.com/GodotVR/godot_oculus_mobile/blob/36393808d159bb7e0a495f7038b0f67790702aa3/src/config/config_common.h#L30

and that only after `bool OvrMobileSession::enter_vr_mode()` is called (https://github.com/GodotVR/godot_oculus_mobile/blob/36393808d159bb7e0a495f7038b0f67790702aa3/src/ovr_mobile_session.cpp#L254)  it gets initialized.

The missing check leads to the call returning unexpected values; but I think this can be fixed by checking and returning an empty godot_variant (maybe with api->godot_variant_new_nil). Or there is a better way to return errors back to the user?

But this led to slighly more complex issue and there should also be a way for the user to check or know if calls are valid because for example at the moment it is not possible to call the Config API in the GDScript
`func _ready():`
after initializing the VR interface but only after the first time
`void OvrMobileSession::process()` is called. Maybe it would be valid to call `bool OvrMobileSession::enter_vr_mode()`already from the initialize() but I don't know the android application lifecycle.

In addition `void OvrMobileSession::exit_vr_mode()` is called when the application goes to sleep; when it wakes up again the first time
`func _process(delta)`
is executed it is again invalid to call the Config classes as the ovr mobile context is still NULL (for the second call it seems to be everything is well intialized again)

### PR DESCRIPTION
**Pull request by m4gr3d**
Add new events and accompanying logic to notify when the app is paused and resumed on Android devices.

One side effect of the pause event on Android is that the egl surface is destroyed and recreated on resume. As such this causes issues for Android apps and/or plugins that promptly makes use of the egl surface on app resume. The new resume event is setup so that it only fires after the app resumes **AND** the egl surface is available.

This helps resolve [GodotVR/godot_oculus_mobile/issue#43](https://github.com/GodotVR/godot_oculus_mobile/issues/43)

Note: I decided to add new events instead of modifying the logic for sending the NOTIFICATION_WM_FOCUS_IN and NOTIFICATION_WM_FOCUS_OUT events in order to provide backward compatibility for existing apps that have no issues with the current logic.

### CODE REVIEW

**Review by Calinou — COMMENTED** [review]
You may want to update the MainLoop documentation to document the new `NOTIFICATION_APP_PAUSED` and `NOTIFICATION_APP_RESUMED` constants :slightly_smiling_face:

See [Contribute to the Class Reference](https://docs.godotengine.org/en/latest/community/contributing/updating_the_class_reference.html).

**Comment by m4gr3d** [comment]
> You may want to update the MainLoop documentation to document the new `NOTIFICATION_APP_PAUSED` and `NOTIFICATION_APP_RESUMED` constants 🙂
> 
> See [Contribute to the Class Reference](https://docs.godotengine.org/en/latest/community/contributing/updating_the_class_reference.html).

Done, thanks for the reminder!

**Review by Calinou — APPROVED** [review]
Documentation looks good to me :slightly_smiling_face:

I can't speak about the implementation itself though, I'll leave that to others.

**Comment by m4gr3d** [comment]
@akien-mga Ping for this PR. It's a blocker for the Godot Oculus mobile plugin (see [PR#44](https://github.com/GodotVR/godot_oculus_mobile/pull/44) and issues [#35](https://github.com/GodotVR/godot_oculus_mobile/issues/35) and [#43](https://github.com/GodotVR/godot_oculus_mobile/issues/43)) so it'd be great if it can make the 3.2 freeze.

**Comment by akien-mga** [comment]
Thanks!

File: doc/classes/MainLoop.xml [hunk 1]
```diff
@@ -215,5 +215,13 @@
 			Notification received from the OS when an update of the Input Method Engine occurs (e.g. change of IME cursor position or composition string).
 			Specific to the macOS platform.
 		</constant>
+		<constant name="NOTIFICATION_APP_RESUMED" value="1014">
```
Comment (akien-mga) [inline doc/classes/MainLoop.xml]: You forgot to bind those constants in MainLoop, so this documentation gets removed. I'll fix it.

**Pull request merged**